trino/auth.py

```diff
@@ -157,8 +161,12 @@ def __eq__(self, other):
 
 
 def handle_redirect_auth_url(auth_url):
-    print("Open the following URL in browser for the external authentication:")
-    print(auth_url)
+    print("Opening your browser for the external authentication:")
+    result = webbrowser.open_new(auth_url)
+    if not result:
```

Comment:
What do you think about having console and browser handlers seperately? This would allow users to use either
```python
connect(
  "coordinator",
  auth=OAuth2Authentication(redirect_auth_url_handler=WebBrowserHandler())
```
or
```python
connect(
  "coordinator",
  auth=OAuth2Authentication(redirect_auth_url_handler=ConsoleHandler())
```

Reply:
I have no objection against having both handlers, but what's the use of the `ConsoleHandler`? If needed, a user can provide their own handler and override the default behaviour of launching the webbrowser. They would probably do something with the url instead of just printing it to the console?

The behaviour is actually pretty similar to how the JDBC driver works. See https://trino.io/docs/current/installation/jdbc.html?highlight=externalAuthentication

Reply:
The jdbc driver actually defines an interface: https://github.com/trinodb/trino/blob/master/client/trino-client/src/main/java/io/trino/client/auth/external/RedirectHandler.java with multiple implementations:
```java
DESKTOP_OPEN(new DesktopBrowserRedirectHandler()),
SYSTEM_OPEN(new SystemOpenRedirectHandler()),
PRINT(new SystemOutPrintRedirectHandler()),
OPEN(new CompositeRedirectHandler(ImmutableList.of(SYSTEM_OPEN, DESKTOP_OPEN))),
ALL(new CompositeRedirectHandler(ImmutableList.of(OPEN, PRINT)))
```
and even though the default is `ALL` users can also use `PRINT` which does not try to open up the browser.

Reply:
Ok, I will adapt the implementation to provide a `ConsoleHandler` and `WebBrowserHandler`.